src/tests/async/awaitingnotasync.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Xunit;
+
+public class AwaitNotAsync
+{
+    [Fact]
+    public static void TestEntryPoint()
+    {
+        AsyncEntryPoint().Wait();
+    }
+
+    private static async Task<T> GetTask<T>(T arg)
+    {
+        await Task.Yield();
+        return arg;
+    }
+
+    // TODO: switch every other scenario to use ValueTask
+    private static async ValueTask<T> GetValueTask<T>(T arg)
+    {
+        await Task.Yield();
+        return arg;
+    }
+
+    private static Task<int> sField;
+
+    private static Task<int> sProp => GetTask(6);
+
+    private static T sIdentity<T>(T arg) => arg;
+
+    private static async2 Task AsyncEntryPoint()
+    {
+        // static field
+        sField = GetTask(5);
+        Assert.Equal(5, await sField);
+
+        // property 
+        Assert.Equal(6, await sProp);
+
+        // generic identity
+        Assert.Equal(6, await sIdentity(sProp));
+
+        // await(await ...))
```

Comment:
this indirectly tests awaiting something that returns T, which happens to be a task by substitution. 
It is ok to use Await in this case now, but will not be optimized into a direct rtAsync call.

Reply:
We do not need to document this, I think. Perhaps one day we may even optimize this...

Reply:
Yeah, this would be interesting to optimize via PGO as some form of guarded deabstraction. I have been thinking about the same thing for delegate calls.